jvector-base/src/main/java/io/github/jbellis/jvector/graph/disk/AbstractGraphIndexWriter.java

```diff
@@ -193,6 +218,58 @@ void writeSparseLevels(ImmutableGraphIndex.View view) throws IOException {
                 throw new IllegalStateException("Mismatch between layer size and nodes written");
             }
         }
+
+        // In V6, fused features for the in-memory hierarchy are written in a block after the top layers of the graph.
+        // Since everything in level 1 is also contained in the high-levels, we only need to write the fused features for level 1.
+        if (version == 6) {
```

Comment:
is this strictly version == 6 or should it be version >= 6 like in another places in the code ?

Reply:
It is a good observation. The rule I followed is that subsequent versions might change this behavior anyway, so there's no benefit in expressing expectations over future versions in the code.